quinn-udp/src/unix.rs

```diff
@@ -91,110 +91,42 @@ fn init(io: SockRef<'_>) -> io::Result<()> {
     // mac and ios do not support IP_RECVTOS on dual-stack sockets :(
     // older macos versions also don't have the flag and will error out if we don't ignore it
     if is_ipv4 || !io.only_v6()? {
-        let on: libc::c_int = 1;
-        let rc = unsafe {
-            libc::setsockopt(
-                io.as_raw_fd(),
-                libc::IPPROTO_IP,
-                libc::IP_RECVTOS,
-                &on as *const _ as _,
-                mem::size_of_val(&on) as _,
-            )
-        };
-        if rc == -1 {
-            tracing::debug!(
-                "Ignoring error setting IP_RECVTOS on socket: {:?}",
-                io::Error::last_os_error()
-            );
+        if let Err(err) = set_socket_option(&*io, libc::IPPROTO_IP, libc::IP_RECVTOS, OPTION_ON) {
+            tracing::debug!("Ignoring error setting IP_RECVTOS on socket: {err:?}",);
         }
     }
 
     #[cfg(target_os = "linux")]
     {
         // opportunistically try to enable GRO. See gro::gro_segments().
-        let on: libc::c_int = 1;
-        unsafe {
-            libc::setsockopt(
-                io.as_raw_fd(),
-                libc::SOL_UDP,
-                libc::UDP_GRO,
-                &on as *const _ as _,
-                mem::size_of_val(&on) as _,
-            )
-        };
+        // TODO: handle error here?
```

Comment:
Per the comment ("opportunistically") ignoring an error is intended here. We want to turn it on if we can, but if we can't that's okay too.

Reply:
👍